index.d.ts

```diff
@@ -75,41 +75,53 @@ type EncodingOption =
 type DefaultEncodingOption = 'utf8';
 type BufferEncodingOption = 'buffer';
 
-type TupleItem<
-	Tuple extends readonly unknown[],
-	Index extends string,
-> = Tuple[Index extends keyof Tuple ? Index : number];
```

Comment:
This type utility created a few issues with other PRs. The goal is for this PR is to get away with it, and replacing it with simpler `extends` clauses.